src/session/SessionContext.jsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * The SessionContext is used to store and retrieve data related to the current user session.
+ */
+const SessionContext = React.createContext({});
+
+/**
+ * Custom hook to retrieve the SessionContext value. Presence of a
+ * value is not validated, as a provider is not required for all applications.
+ * @returns The retrieved SessionContext value.
+ */
+const useSession = () => React.useContext(SessionContext);
+
+const sessionContextShape = PropTypes.shape({
+  /**
+   * Whether or not the current session is active or in an inactive state i.e. "locked".
+   */
+  isActive: PropTypes.bool,
+  /**
+   * An object used to store data about the session and its user.
+   */
+  metaData: PropTypes.object,
+  /**
+   * An object representing the user attributed to the current session.
+   */
+  user: PropTypes.shape({
+    /**
+     * The user's system username.
+     */
+    username: PropTypes.string,
+    /**
+     * The user's first name.
+     */
+    firstName: PropTypes.string,
+    /**
+     * The user's last name.
+     */
+    lastName: PropTypes.string,
```

Comment:
This looks great. Wondering if we want to include a separate metaData property in the use shape, or if we'd expect that to roll up into the root-level metaData object?

Reply:
I'd be fine with either, depends on how easy it is to message the expectation to users.

Reply:
I imagine it's fine as it is. I can see us adding more first-class keys to user over time and relying on root metaData for anything else.